governator-core/src/main/java/com/netflix/governator/InjectorBuilder.java

```diff
@@ -0,0 +1,221 @@
+package com.netflix.governator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.Binder;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import com.google.inject.Stage;
+import com.google.inject.spi.Element;
+import com.google.inject.spi.ElementVisitor;
+import com.google.inject.spi.Elements;
+import com.google.inject.util.Modules;
+import com.netflix.governator.spi.InjectorCreator;
+import com.netflix.governator.spi.ModuleTransformer;
+import com.netflix.governator.visitors.IsNotStaticInjectionVisitor;
+import com.netflix.governator.visitors.KeyTracingVisitor;
+import com.netflix.governator.visitors.WarnOfStaticInjectionVisitor;
+
+/**
+ * Simple DSL on top of Guice through which an injector may be created using a series
+ * of operations and transformations of Guice modules.  Operations are tracked using a 
+ * single module and are additive such that each operation executes on top of the entire 
+ * current binding state.  Once all bindings have been defined the injector can be created 
+ * using an {@link InjectorCreator} strategy.
+ * 
+ * <code>
+ * InjectorBuilder
+ *      .fromModule(new MyApplicationModule())
+ *      .overrideWith(new OverridesForTesting())
+ *      .traceEachElement(new BindingTracingVisitor())
+ *      .createInjector();
+ * </code>
+ */
+public final class InjectorBuilder {
+    private static final Logger LOG = LoggerFactory.getLogger(InjectorBuilder.class);
+    
+    private static final Stage LAZY_SINGLETONS_STAGE = Stage.DEVELOPMENT;
+    
+    private Module module;
+    
+    /**
+     * Start the builder using the specified module. 
+     * 
+     * @param module
+     * @return
+     */
+    public static InjectorBuilder fromModule(Module module) {
+        return new InjectorBuilder(module);
+    }
+    
+    public static InjectorBuilder fromModules(Module firstModule, Module secondModule, Module ... additionalModules) {
+        List<Module> modules = new ArrayList<>();
+        modules.add(firstModule);
+        modules.add(secondModule);
+        if (null != additionalModules) {
+            modules.addAll(Arrays.asList(additionalModules));
+        }
+        return new InjectorBuilder(Modules.combine(modules));
+    }
+    
+    public static InjectorBuilder fromModules(List<Module> modules) {
+        return new InjectorBuilder(Modules.combine(modules));
+    }
+    
+    public static InjectorBuilder createDefault() {
```

Comment:
Is there a point to this method? Would a user ever try to create an injector without at last one module to start with?